packages/components/src/internal/components/labels/LabelsConfigurationPanel.tsx

```diff
@@ -0,0 +1,352 @@
+    const { api, setIsDirty } = props;
+    const [templates, setTemplates] = useState<LabelTemplate[]>([]);
+    const [error, setError] = useState<string>();
+    const [selected, setSelected] = useState<number>();
+    const addNew = useMemo(() => selected === NEW_LABEL_INDEX, [selected]);
+
+    const queryLabelTemplates = useCallback(
+        (newLabelTemplate?: number) => {
+            setError(undefined);
+
+            api.labelprinting
+                .getLabelTemplates()
+                .then(labelTemplates => {
+                    setTemplates(labelTemplates);
+
+                    if (newLabelTemplate)
+                        setSelected(labelTemplates.findIndex(template => template.rowId === newLabelTemplate));
+                })
+                .catch(reason => {
+                    if (reason.status === 404) {
+                        // try to create list
+                        api.labelprinting
+                            .ensureLabelTemplatesList()
+                            .then(() => queryLabelTemplates(newLabelTemplate))
+                            .catch(newReason => {
+                                console.error(newReason);
+                                setError('Error: Failed to create template list.');
+                            });
+                    } else {
+                        console.error(reason);
+                        setError('Error: Unable to load label templates.');
+                    }
+                });
+        },
+        [api]
+    );
+
+    // Load template list
+    useEffect(() => {
+        queryLabelTemplates();
+    }, [queryLabelTemplates]);
+
+    const onSetSelected = useCallback((index: number) => {
+        setSelected(index);
+    }, []);
+
+    const onAddLabel = useCallback(() => {
+        setSelected(NEW_LABEL_INDEX);
+    }, []);
+
+    const onChange = useCallback(() => {
+        setIsDirty(true);
+    }, [setIsDirty]);
+
+    const onActionCompleted = useCallback(
+        (newLabelTemplate?: number, isDelete = false): void => {
+            queryLabelTemplates(newLabelTemplate);
+            if (isDelete) setSelected(undefined);
+            setIsDirty(false);
+        },
+        [queryLabelTemplates, setIsDirty]
+    );
+
+    return (
+        <div className="panel panel-default label-templates-container">
+            <div className="list__bold-text">{TITLE}</div>
+            <div>
+                {error && <Alert>{error}</Alert>}
+                {!templates && <LoadingSpinner />}
+                {templates && !error && (
+                    <div className="row choices-container">
+                        <div className="col-lg-4 col-md-6 choices-container-left-panel">
+                            <LabelTemplatesList templates={templates} selected={selected} onSelect={onSetSelected} />
+                            <AddEntityButton onClick={onAddLabel} entity="New Status" disabled={addNew} />
```

Comment:
The button currently says "Add New Status" should that say "Add New Label Template" or something along those lines?

Reply:
done